crates/sqlexec/src/context/local.rs

```diff
@@ -84,7 +84,9 @@ impl LocalSessionContext {
         conf = conf
             .with_extension(Arc::new(catalog_mutator))
             .with_extension(Arc::new(native_tables.clone()))
-            .with_extension(Arc::new(TempCatalog::default()));
+            .with_extension(Arc::new(TempCatalog::default()))
+            .with_extension(Arc::new(catalog.clone()));
```

Comment:
@scsmithr I know we originally removed the `catalog` from the extension as we only needed the mutator for the DDL's, but now we need the catalog again. 

I could move the logic that uses the catalog into the `session_planner`, but it seemed more appropriate to have it in the `DescribeTableExec`. 

any thoughts on this approach?

Reply:
We could also split this up into multiple execs `[DescribeNativeTableExec, DescribeBuiltinTableExec, DescribeTempTableExec]`. 

This would make it so the execs don't need the full catalog, but it seemed simpler to just pass in the catalog through the extensions. If there's a good reason to not have the catalog on the extension, i can refactor this into multiple execs.

Reply:
What's stopping us from putting a `TableEntry` on the Describe exec instead of the `OwnedFullObjectReference`?

We're already getting table entry here: https://github.com/glaredb/glaredb/blob/b0e3334fb23b86e90a14908ae945a02a2f6015bc/crates/sqlexec/src/planner/session_planner.rs#L668-L670

And so we can just stick that on the describe plans and reference that directly in the exec without needing to reference the catalog.

We have a couple other execs that already use `TableEntry`, e.g. updates:
- https://github.com/glaredb/glaredb/blob/b0e3334fb23b86e90a14908ae945a02a2f6015bc/crates/protogen/src/sqlexec/physical_plan.rs#L261-L269
- https://github.com/glaredb/glaredb/blob/b0e3334fb23b86e90a14908ae945a02a2f6015bc/crates/sqlexec/src/planner/physical_plan/update.rs#L20-L25

Reply:
oh good idea. I didn't notice `TableEntry` was serializable.